spec/@esm/esms_util_number_to_string_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
```

Comment:
Is this an example of test driven development? 
What sort of google search would I use to figure out the methodology you used here and then implement it for my own projects where appropriate (ie, the very little automated testing I did in java was with JUnit)

Reply:
RSpec is an example of a testing framework for Ruby, like JUnit is for Java. I started with RSpec but there are other testing frameworks out there that solve different use cases such as Minitest, Cucumber, unit-test, etc. Each one has its benefits and and disadvantages but I don't think you'd be wrong using any of them. Minitest and RSpec being the most popular because of Rails. 

As for TDD, y understanding is this: "it's a workflow".
Instead of writing your code and manually testing it along the way, or not testing it at all, TDD forces you to write the tests first and then you write your code. 

For example, you want to add a button to a site? Write a test to check for the button and watch it fail. Then add the button and run the test. Tada! 

_If someone says they follow, or strictly follow, TDD don't believe them and ask for them to prove it lol. Ain't nobody got time for that_